### PR TITLE
use redemption location for merchants with no lat/lng

### PR DESCRIPTION
**Pull request by aleqi200**
Sample Merchant: https://api.groupon.com/v2/merchants/jv-beauty-skin-care
Sample Deal: http://api.groupon.com/v2/deals/a-touch-of-color
As last resort we can look up using the division.
Also we could use the redemption address.

One TODO is to handle multiple locations.

### CODE REVIEW

File: app/src/main/java/com/grelp/grelp/activities/GrouponDetailActivity.java [hunk 1]
```diff
@@ -75,7 +79,25 @@ public void onFailure(int statusCode, Header[] headers, Throwable throwable, JSO
     }
 
     private void searchForMerchantOnYelp() {
-        YelpAPI.getInstance().searchForBusinesses(merchant.getName(), merchant.getLat(), merchant.getLng(),
+        double lat = merchant.getLat();
+        double lng = merchant.getLng();
+        LinkedList<RedemptionLocation> locations = new LinkedList<>(groupon.getUniqueRedemptionLocations());
```
Comment (vgudla) [inline app/src/main/java/com/grelp/grelp/activities/GrouponDetailActivity.java:84]: Use interface :)

**Comment by vgudla** [comment]
Thank you for adding the fallback. 
LGTM.

File: app/src/main/java/com/grelp/grelp/activities/GrouponDetailActivity.java (hunk 1 above)
Comment (aleqi200) [inline app/src/main/java/com/grelp/grelp/activities/GrouponDetailActivity.java:84]: using lnked list for the convenience of getFirst instead of get(0)

Comment (vgudla) [inline app/src/main/java/com/grelp/grelp/activities/GrouponDetailActivity.java:84]: Ah didn't see that

**Pull request merged**